core/src/main/java/org/apache/carbondata/core/scan/result/BlockletScannedResult.java

```diff
@@ -530,6 +530,20 @@ protected void fillValidRowIdsBatchFilling(int rowId, int batchSize) {
     }
   }
 
+  protected void fillValidRowIdsBatchFillingWithFilter(int rowId, int batchSize) {
+    // row id will be different for every batch so clear it before filling
+    clearValidRowIdList();
+    int startPosition = rowId;
+    for (int i = 0, j = startPosition;
+         i < batchSize && j < pageFilteredRowId[pageCounter].length; i++) {
```

Comment:
while this i<batchSize is required, i think caller will handle batchSize will be minimum(batchsize,number of rows in page)...can u please check

Reply:
Yes I tested and the caller handles it so removed.